### DISCUSSION TITLE
Some CAD features

### DISCUSSION THREAD

**Discussion by ghost** [SVG-Edit/svgedit#797] [New feature request]
**Feature Request**
1. Snap points: on line, on circle, polygon corners, midle|start|end of line, start and end of path(directly from line). Basicly needs any snap interface to near objects - with drawing points connected after move oeration.(base function)
2. Linking objects - ortoganal connecting with line(base function)
3. Labels with line to object i.e. CAD position mark of detail in assembly drawing (plugin function)

I am think use svg-edit as some free online CAD system.

I am plan to do this modification by myself, but need some help.

**Comment by haisenbergX** [comment]
have the same problem with  CAD position mark

what you do this

**Comment by ghost** [comment]
I am cre #319 for that.
It is first version and have only base functions and some bugs not fixed yet.

**Comment by ghost** [comment]
How snap x,y to gird? I found only bool function ```svgCanvas.getSnapToGrid()```
mouseDown() get alredy snapped, but mouseMove() is not
```javascript
          const scale = 1/svgCanvas.getZoom();
          let x = opts.mouse_x*scale;
          let y = opts.mouse_y*scale;
          /*if(svgCanvas.getSnapToGrid()){
              //TODO: Snap to gird.
          }*/
```
UPD: It started with snapped position and i am think - move snap do not needs.
